### PR TITLE
New: Rollup model, serializers and consumers

### PR DESCRIPTION
**Pull request by cythrawll**
## Description

Adds rollup fragment for fili subtotal feature

## Proposed Changes

- Add rollup fragment and add it to request model with supporting methods
- Add feature flag and remove rollup from serialize rollup if it's off
- Adjust dimension transform so dateTime works correctly
- Add the rollup consumers

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: packages/core/addon/models/bard-request/request.js [hunk 1]
```diff
@@ -398,6 +399,33 @@ export default Fragment.extend(Validations, {
     filter.setProperties(props);
   },
 
+  /* == Rollup == */
+
+  pushRollupColumn(dimensionObj) {
+    if (!this.rollup.columns) {
+      this.rollup = this.store.createFragment('bard-request/fragments/rollup');
+    }
```
Comment (kevinhinterlong) [inline packages/core/addon/models/bard-request/request.js:407]: is this needed with `defaultValue`?

File: packages/core/addon/models/bard-request/request.js [hunk 2]
```diff
@@ -398,6 +399,33 @@ export default Fragment.extend(Validations, {
...
+    this.removeRollupColumn(dimensionObj); //remove duplicate
+    this.rollup.columns.createFragment(dimensionObj);
+  },
+
+  removeRollupColumn(dimensionObj) {
+    if (!this.rollup.columns) {
+      this.rollup = this.store.createFragment('bard-request/fragments/rollup');
+    }
+    const columns = this.rollup.columns,
+      dimension = dimensionObj.dimension;
```
Comment (kevinhinterlong) [inline packages/core/addon/models/bard-request/request.js:417]:
```suggestion
    const { columns } = this.rollup;
    const { dimension } = dimensionObj;
```

File: packages/core/tests/unit/models/bard-request/request-test.js [hunk 3]
```diff
@@ -1852,4 +1852,65 @@ module('Unit | Model Fragment | BardRequest - Request', function(hooks) {
...
   });
+
+  test('Rollup request features', assert => {
+    assert.expect(5);
+    let mockModel = Store.peekRecord('fragments-mock', 1),
+      request = mockModel.get('request');
+
+    request.addDimension({ dimension: 'bar' });
+
+    request.pushRollupColumn({ dimension: 'foo' });
```
Comment (kevinhinterlong) [inline packages/core/tests/unit/models/bard-request/request-test.js:1863]: test duplicate

File: packages/data/tests/dummy/mirage/config.js [hunk 4]
```diff
@@ -1,5 +1,6 @@
 import config from 'dummy/config/environment';
 import BardMeta from 'navi-data/mirage/routes/bard-meta';
+import BardLight from 'navi-data/mirage/routes/bard-lite';
```
[the remark this answers is not in the document]
Comment (cythrawll) [reply packages/data/tests/dummy/mirage/config.js:3]: I think I was having a beer when I typed that.

**Review by kevinhinterlong — APPROVED** [review]

File: packages/core/addon/models/bard-request/request.js [hunk 5]
```diff
@@ -83,6 +83,7 @@ export default Fragment.extend(Validations, {
     defaultValue: []
   }),
   dataSource: DS.attr('string', { defaultValue: getDefaultDataSourceName() }),
+  rollup: fragment('bard-request/fragments/rollup', { defaultValue: { columns: [], grandTotal: false } }),
```
Comment (jkusa) [inline packages/core/addon/models/bard-request/request.js:86]:
```suggestion
  rollup: fragment('bard-request/fragments/rollup', { defaultValue: () => ({ columns: [], grandTotal: false } })),
```
??

**Review by jkusa — APPROVED** [review]
Just the one suggestion, otherwise 👍

**Pull request merged**